### PR TITLE
JWT Authentication

### PR DESCRIPTION
**Pull request by chrisvrose**
Add JWT Bearer authentication support. Http Basic auth will be continued

Two APIs have been added:
1. POST `/api/auth` for login
2. DELETE `/api/auth` for logout

Tokens created are validated on the backend, and also checked against db list of tokens for user.

It is unfortunate that we store it on backend, but it is done to allow for a "logout"

### CODE REVIEW

**Comment by chrisvrose** [comment]
This is a wordy PR. This will require some time.

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1 +1,2 @@
 # spring.config.import=optional:file:.env[.properties]
+jwt.secret=1abcdascabADGAGEbcdabc531dab1235cd!@)*$(abcd
```
Comment (Prajwalprakash3722) [inline src/main/resources/application.properties:2]: Hmm... secret out lol

**Review by Prajwalprakash3722 — APPROVED** [review]
LGTM, looks like awfully lot of code just to implement JWT.

File: src/main/resources/application.properties (hunk 1 above)
Comment (chrisvrose) [reply src/main/resources/application.properties:2]: Yep lmao
Shall replace it for actual deployment

**Pull request merged**